## Preliminaries

#### Datetime

In [15]:
from datetime import datetime as dt 
now = dt.now()
print("Analysis on", now.strftime('%Y-%m-%d'), "at", now.strftime('%H:%M %p'))

Analysis on 2023-08-09 at 21:10 PM


#### Establish CWD

Identifying the current working directory from which the data is stored:

In [16]:
import os 
os.getcwd()

'/Users/chasecarlson/Documents/GSCM Course Materials/GSCM 575 Machine Learning in Business/Python Pjojects/GSCM-575-ML/code'

#### Import libraries

Import the following core libraries to support the analysis:

In [17]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

## About the Data

#### About Dataset

This dataset created from a higher education institution (acquired from several disjoint databases) related to students enrolled in different undergraduate degrees, such as agronomy, design, education, nursing, journalism, management, social service, and technologies. The dataset includes information known at the time of student enrollment (academic path, demographics, and social-economic factors) and the students' academic performance at the end of the first and second semesters. The data is used to build classification models to predict students' dropout and academic success. The problem is formulated as a three-category classification task, in which there is a strong imbalance towards one of the classes.

#### Data Attributes

| Column name                  | Description                                                                                          |
|-----------------------------|------------------------------------------------------------------------------------------------------|
| Marital status              | The marital status of the student. (Categorical)                                                    |
| Application mode            | The method of application used by the student. (Categorical)                                        |
| Application order           | The order in which the student applied. (Numerical)                                                 |
| Course                      | The course taken by the student. (Categorical)                                                      |
| Daytime/evening attendance  | Whether the student attends classes during the day or in the evening. (Categorical)                |
| Previous qualification      | The qualification obtained by the student before enrolling in higher education. (Categorical)        |
| Nationality                 | The nationality of the student. (Categorical)                                                        |
| Mother's qualification      | The qualification of the student's mother. (Categorical)                                             |
| Father's qualification      | The qualification of the student's father. (Categorical)                                             |
| Mother's occupation         | The occupation of the student's mother. (Categorical)                                                |
| Father's occupation         | The occupation of the student's father. (Categorical)                                                |
| Displaced                   | Whether the student is a displaced person. (Categorical)                                             |
| Educational special needs   | Whether the student has any special educational needs. (Categorical)                                 |
| Debtor                      | Whether the student is a debtor. (Categorical)                                                        |
| Tuition fees up to date     | Whether the student's tuition fees are up to date. (Categorical)                                     |
| Gender                      | The gender of the student. (Categorical)                                                              |
| Scholarship holder          | Whether the student is a scholarship holder. (Categorical)                                            |
| Age at enrollment           | The age of the student at the time of enrollment. (Numerical)                                         |
| International               | Whether the student is an international student. (Categorical)                                        |
| Curricular units 1st sem (credited)   | The number of curricular units credited by the student in the first semester. (Numerical)       |
| Curricular units 1st sem (enrolled)   | The number of curricular units enrolled by the student in the first semester. (Numerical)       |
| Curricular units 1st sem (evaluations) | The number of curricular units evaluated by the student in the first semester. (Numerical)     |
| Curricular units 1st sem (approved)    | The number of curricular units approved by the student in the first semester. (Numerical)      |

#### Import Dataset

In [18]:
df = pd.read_csv('data/student_success.csv')
df.head().transpose()

0          1        2  \
Marital status                                        1          1        1   
Application mode                                      8          6        1   
Application order                                     5          1        5   
Course                                                2         11        5   
Daytime/evening attendance                            1          1        1   
Previous qualification                                1          1        1   
Nacionality                                           1          1        1   
Mother's qualification                               13          1       22   
Father's qualification                               10          3       27   
Mother's occupation                                   6          4       10   
Father's occupation                                  10          4       10   
Displaced                                             1          1        1   
Educational special needs                             0          0        0   
Debtor                                                0          0        0   
Tuition fees up to date                               1          0        0   
Gender                                                1          1        1   
Scholarship holder                                    0          0        0   
Age at enrollment                                    20         19       19   
International                                         0          0        0   
Curricular units 1st sem (credited)                   0          0        0   
Curricular units 1st sem (enrolled)                   0          6        6   
Curricular units 1st sem (evaluations)                0          6        0   
Curricular units 1st sem (approved)                   0          6        0   
Curricular units 1st sem (grade)                    0.0       14.0      0.0   
Curricular units 1st sem (without evaluations)        0          0        0   
Curricular units 2nd sem (credited)                   0          0        0   
Curricular units 2nd sem (enrolled)                   0          6        6   
Curricular units 2nd sem (evaluations)                0          6        0   
Curricular units 2nd sem (approved)                   0          6        0   
Curricular units 2nd sem (grade)                    0.0  13.666667      0.0   
Curricular units 2nd sem (without evaluations)        0          0        0   
Unemployment rate                                  10.8       13.9     10.8   
Inflation rate                                      1.4       -0.3      1.4   
GDP                                                1.74       0.79     1.74   
Target                                          Dropout   Graduate  Dropout   

                                                        3          4  
Marital status                                          1          2  
Application mode                                        8         12  
Application order                                       2          1  
Course                                                 15          3  
Daytime/evening attendance                              1          0  
Previous qualification                                  1          1  
Nacionality                                             1          1  
Mother's qualification                                 23         22  
Father's qualification                                 27         28  
Mother's occupation                                     6         10  
Father's occupation                                     4         10  
Displaced                                               1          0  
Educational special needs                               0          0  
Debtor                                                  0          0  
Tuition fees up to date                                 1          1  
Gender                                                  0          0  
Sc

View the dimensions of the data:

In [19]:
df.shape

(4424, 35)

The data frame has 4,424 rows and 35 columns.

## Data Preprocessing

View basic statistics:

In [20]:
df.describe().transpose()

count       mean        std  \
Marital status                                  4424.0   1.178571   0.605747   
Application mode                                4424.0   6.886980   5.298964   
Application order                               4424.0   1.727848   1.313793   
Course                                          4424.0   9.899186   4.331792   
Daytime/evening attendance                      4424.0   0.890823   0.311897   
Previous qualification                          4424.0   2.531420   3.963707   
Nacionality                                     4424.0   1.254521   1.748447   
Mother's qualification                          4424.0  12.322107   9.026251   
Father's qualification                          4424.0  16.455244  11.044800   
Mother's occupation                             4424.0   7.317812   3.997828   
Father's occupation                             4424.0   7.819168   4.856692   
Displaced                                       4424.0   0.548373   0.497711   
Educational special needs                       4424.0   0.011528   0.106760   
Debtor                                          4424.0   0.113698   0.317480   
Tuition fees up to date                         4424.0   0.880651   0.324235   
Gender                                          4424.0   0.351718   0.477560   
Scholarship holder                              4424.0   0.248418   0.432144   
Age at enrollment                               4424.0  23.265145   7.587816   
International                                   4424.0   0.024864   0.155729   
Curricular units 1st sem (credited)             4424.0   0.709991   2.360507   
Curricular units 1st sem (enrolled)             4424.0   6.270570   2.480178   
Curricular units 1st sem (evaluations)          4424.0   8.299051   4.179106   
Curricular units 1st sem (approved)             4424.0   4.706600   3.094238   
Curricular units 1st sem (grade)                4424.0  10.640822   4.843663   
Curricular units 1st sem (without evaluations)  4424.0   0.137658   0.690880   
Curricular units 2nd sem (credited)             4424.0   0.541817   1.918546   
Curricular units 2nd sem (enrolled)             4424.0   6.232143   2.195951   
Curricular units 2nd sem (evaluations)          4424.0   8.063291   3.947951   
Curricular units 2nd sem (approved)             4424.0   4.435805   3.014764   
Curricular units 2nd sem (grade)                4424.0  10.230206   5.210808   
Curricular units 2nd sem (without evaluations)  4424.0   0.150316   0.753774   
Unemployment rate                               4424.0  11.566139   2.663850   
Inflation rate                                  4424.0   1.228029   1.382711   
GDP                                             4424.0   0.001969   2.269935   

                                                  min    25%        50%  \
Marital status                                   1.00   1.00   1.000000   
Application mode                                 1.00   1.00   8.000000   
Application order                                0.00   1.00   1.000000   
Course                                           1.00   6.00  10.000000   
Daytime/evening attendance                       0.00   1.00   1.000000   
Previous qualification                           1.00   1.00   1.000000   
Nacionality                                      1.00   1.00   1.000000   
Mother's qualification                           1.00   2.00  13.000000   
Father's qualification                           1.00   3.00  14.000000   
Mother's occupation                              1.00   5.00   6.000000   
Father's occupation                              1.00   5.00   8.000000   
Displaced                                        0.00   0.00   1.000000   
Educational special needs                        0.00   0.00   0.000000   
Debtor                                           0.00   0.00   0.000000   
Tuition fees up to date                          0.00   1.00   1.000000   
Gender                                          

Display the data types for each variable:

In [21]:
df.dtypes

Marital status                                      int64
Application mode                                    int64
Application order                                   int64
Course                                              int64
Daytime/evening attendance                          int64
Previous qualification                              int64
Nacionality                                         int64
Mother's qualification                              int64
Father's qualification                              int64
Mother's occupation                                 int64
Father's occupation                                 int64
Displaced                                           int64
Educational special needs                           int64
Debtor                                              int64
Tuition fees up to date                             int64
Gender                                              int64
Scholarship holder                                  int64
Age at enrollm